src/util/charbuf.c

```diff
@@ -136,3 +136,12 @@ charbuf copy_chars_from_charbuf(charbuf buf, size_t index)
   }
   return new_charbuf(0);
 }
+
+unsigned char * null_terminated_string(charbuf buf)
```

Comment:
This function should check that ```buf.chars``` is not ```NULL``` and ```buf.len > 0``` before it does any allocation or copying.